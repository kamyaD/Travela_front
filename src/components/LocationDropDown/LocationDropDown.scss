.locationbar {
  &__location {
    margin: 0 15px;
    align-items: center;
    justify-content: center;
    @media (max-width: 479px) {
      margin: 0;
      width: 100%;
    }
  }

  &__location input,
  &__location input:active,
  &__location input:hover,
  &__location input:focus {
    outline: none;
  }

  &__location-box {
    height: 36px;
    border-radius: 5px;
    background-color: #f8f8f8;
    margin-top: 15px;
    @media (max-width: 479px) {
      display: flex !important;
      align-items: center;
      justify-content: center;
      margin: 5px 0 10px !important;
    }
  }

  &__location,
  &__location input {
    /* Cross browser flex-grow */
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
  }

  &__location,
  &__location form {
    /* Cross browser inline-flex */
    display: -webkit-inline-box;
    display: -webkit-inline-flex;
    display: -moz-inline-box;
    display: -ms-inline-flexbox;
    display: inline-flex;

    @media (max-width: 781px) {
      width: 210px;
    }
  }

  /* Position search icon */
  &__location-icon {
    margin-right: 7px;
    cursor: pointer;
    @media (max-width: 781px) {
      left: 190px;
    }
    @media (max-width: 612px) {
      display: none;
    }
  }

  &__location input {
    border: none;
    border-radius: 18px;
    font-size: 16px;
    color: #999999;
    background-color: #f8f8f8;
    padding: 0px 5px 0px 20px;
    line-height: 18px;
    font-family: "DIN Pro";
    font-size: 14px;
    height: 36px;
    width: 100px;

    @media (max-width: 781px) {
      font-size: xx-small;
    }
  }

  &__location input::placeholder {
    color: #999999;
  }
}

.dropDown {
  float: left;
  overflow: hidden;
}

.dropDown .dropBtn {
  font-size: 16px;
  border: none;
  outline: none;
  background-color: inherit;
  font-family: "DIN Pro";
  margin: 0;
  cursor: pointer;
}

.dropDown-content {
  display: none;
  position: absolute;
  background-color: white;
  width: 147px;
  z-index: 1;
  padding: 0;
}

.content {
  float: none;
  color: black;
  padding: 12px 16px;
  text-decoration: none;
  display: block;
  text-align: left;
}

.dropDown-hover:hover {
  background-color: #f8f8f8;
  cursor: pointer;
}

.location_line {
  box-sizing: border-box;
  height: 34px;
  width: 1px;
  border: 1px solid rgba(178, 187, 198, 0.37);
  margin: 0px 0px -33px 9px;
}

.location_circle {
  width: 17px;
  height: 10px;
  background: #f8f8f8;
  border-radius: 100px / 50px;
  border: 2px solid #B2BBC6;
  margin: 13px 0px 0px 11px;
}


.location_circle_inner {
  height: 2.5px;
  width: 5px;
  background: #f8f8f8;
  border-radius: 100px / 50px;
  border: 2px solid #B2BBC6;
  margin: -10px 0px 0px 17px;
}
