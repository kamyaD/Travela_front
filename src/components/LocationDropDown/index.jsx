import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import selectDropDownIcon from '../../images/icons/form_select_dropdown.svg';
import './LocationDropDown.scss';

class LocationDropDown extends PureComponent {
  state = {
    dropDown: false,
    location: 'All Locations',
  }

  static getDerivedStateFromProps(nextProps, prevState){
    if(nextProps.urlLocation.search  === '') return {location: 'All Locations'};
    return null;
  }

  handleDropDown = (e) => {
    e.preventDefault;
    const {dropDown} = this.state;
    this.setState({ dropDown: !dropDown});
  }

  selectDropDown = (e, center) => {
    e.preventDefault;
    const { getCenter } = this.props;
    this.setState({ location: center });
    getCenter(center);
    localStorage.setItem('center', center);
  }
  
  render() {
    let { dropDown, location } = this.state;
    const { centers, urlLocation } = this.props;
    const newCenter = centers.slice();
    const showDropDown = ['/requests/my-verifications', '/dashboard', '/trip-planner/checklists'];
    const { pathname } = urlLocation;
    location !== 'All Locations' ?  newCenter.push('All Locations') : null;
    return (
      <div>
        {
          showDropDown.includes(pathname) ?
            (
              <div className="mdl-search locationbar__location">

                <div className="locationbar__location-box">
                  <div className="location_circle" />
                  <div className="location_circle_inner" />
                </div>
      
                <div 
                  className="locationbar__location-box" 
                  role="presentation"
                  onKeyDown={this.keyPress}
                  onClick={this.handleDropDown}>
                  <div className="location_line" />

                  <div className="dropDown">
                    <input
                      id="search"
                      className="dropBtn"
                      type="text"
                      disabled
                      placeholder={location}
                    />
                    <img 
                      src={selectDropDownIcon} 
                      alt="icn" 
                      className="locationbar__location-icon" />
                    <ul className="dropDown-content" style={{ display: dropDown ? 'block': null }}>
                      {
                        newCenter.map((center) => 
                          (
                            <li
                              className="content dropDown-hover" 
                              onKeyDown={this.keyPress}
                              role="presentation"
                              onClick={e => this.selectDropDown(e, center)} 
                              key={center}>
                              {center}
                            </li>)
                        )
                      }
                    </ul>
                  </div>
                </div>
              </div>
            ): null
        }
      </div>
    );
  }
}

LocationDropDown.propTypes = {
  centers:  PropTypes.array.isRequired,
  urlLocation: PropTypes.shape({}).isRequired,
  getCenter: PropTypes.func.isRequired,
};
export default LocationDropDown;
