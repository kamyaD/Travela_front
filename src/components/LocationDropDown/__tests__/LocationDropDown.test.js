import React from 'react';
import { shallow, mount } from 'enzyme';
import LocationDropDown from '../index';

class MockLocalStorage {
  constructor() {
    this.data = {};
  }

  setItem(item, value) {
    this.data[item] = value.toString();
  }
}

global.localStorage = new MockLocalStorage();

const props = {
  centers: ['Togo'],
  urlLocation: {
    pathname: '/requests/my-verifications'
  },
  getCenter: jest.fn()
};

const wrapper = mount(<LocationDropDown {...props} />);

describe('Render Location DropDown component', () => {

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the Location DropDown as expected', () => {
    expect(wrapper.length).toBe(1);
  });

  it('should call handleDropDown when the drop down  is triggered', () => {
    const spy = jest.spyOn(wrapper.instance(), 'selectDropDown');
    wrapper.find('.dropBtn').simulate('click');
    wrapper.find('.content').simulate('click');
    expect(spy).toHaveBeenCalledTimes(1);
  });
});
