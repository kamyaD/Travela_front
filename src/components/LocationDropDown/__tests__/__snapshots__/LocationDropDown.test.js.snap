// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Location DropDown component should match snapshot 1`] = `
<div>
  <div
    className="mdl-search locationbar__location"
  >
    <div
      className="locationbar__location-box"
    >
      <div
        className="location_circle"
      />
      <div
        className="location_circle_inner"
      />
    </div>
    <div
      className="locationbar__location-box"
      onClick={[Function]}
      role="presentation"
    >
      <div
        className="location_line"
      />
      <div
        className="dropDown"
      >
        <input
          className="dropBtn"
          disabled={true}
          id="search"
          placeholder="All Locations"
          type="text"
        />
        <img
          alt="icn"
          className="locationbar__location-icon"
          src="form_select_dropdown.svg"
        />
        <ul
          className="dropDown-content"
          style={
            Object {
              "display": null,
            }
          }
        >
          <li
            className="content dropDown-hover"
            key="Togo"
            onClick={[Function]}
            role="presentation"
          >
            Togo
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
